### ISSUE TITLE
[2-coverage] Def-clearness with sidetrips

### ISSUE THREAD

**Issue by Allgot** [swtv-kaist/cs458-spring23#6]
Hello!

I am curious how we should define "def-clearness" with sidetrips.

For the sake of easier understand, let me just define a graph first:
```
N = {0, 1, 2, 3}
N_0 = {0}  
N_f = {3}  
E = {(0, 1), (1, 2), (1, 3), (2,3)}  
def(0) = def(2) = use(3) = {x}  
```
That is,

![image](https://user-images.githubusercontent.com/96973603/226298275-fb43a484-c6b5-4d88-8cd1-bcaa233f4ffb.png)

In this graph, [0, 1, 3] is du-path. And if we allow sidetrips, [0, 1, 2, 3] **tours** this [0, 1, 3] path, but I'm not sure it also **du-tours** this path. The point I'm comfusing is the fact that although [0, 1, 2, 3] defines x twice so it is not def-clear, but still the subpath [0, 1, 3] contains only one definition of x (since 2 is just a sidetrip). 

In this case, can we say that [0, 1, 2, 3] du-tours [0, 1, 3]?

In other words, should we consider sidetrips as well when determining def-clearness?

Thank you for reading!

**Comment by Allgot** [comment]
Btw, it seems I cannot label my issue by myself. Would any maintainer of this repository check this issue too?

Thanks :)

**Comment by swtv-kaist** [comment]
> In this case, can we say that [0, 1, 2, 3] du-tours [0, 1, 3]?\

No.  [0,1,2,3] does not du-tours [0,1,3] since it does not exercise data flow of x from node 0 to node 3.

**Comment by Allgot** [comment]
So, we should consider sidetrips as well when it comes to def-clearness!

Thank you!

**Comment by jaehyun1ee** [comment]
BTW, I think [0, 1, 2, 3] doesn't tour [0, 1, 3] with sidetrips, because we have defined sidetrips as,

```
A test path p tours subpath q with sidetrips iff every edge in q is also in p in the same order.
```

The test path [0, 1, 2, 3] misses the edge (1, 3) in [0, 1, 3].

**Comment by Allgot** [comment]
ahh thanks for pointing it!

I've made a wrong example, but still prof. gave the right answer for my question 👍

**Comment by TonyWild** [comment]
I'm slightly confused by the professor's answer.
The question was "whether [0, 1, 2, 3] du-tours [0, 1, 3]", and the answer was "No. [0, 1, 2, 3] is not a du-path".

Ignoring the fact that [0, 1, 2, 3] actually does not tour with sidetrip [0, 1, 3],
when we say "a test path p du-tours a subpath d", does p itself have to be a du-path (i.e., def-clear)?

According to material,
`p du-tours d with respect to v if p tours d and the subpath taken is def-clear with respect to v`
Here, what does "the subpath taken" mean? Is it p or d?

**Comment by swtv-kaist** [comment]
@TonyWild:
You are right.  My previous comment was misleading.  I modified my previous comment to clarify the meaning.

> ``` p du-tours d with respect to v if p tours d and the subpath taken is def-clear with respect to v```
> Here, what does "the subpath taken" mean? Is it p or d?

"the subpath taken" is d
